src/SessionManager/ProgramSettings.cc

```diff
@@ -182,6 +270,25 @@ sending messages to the backend).
             files.push_back(relative_path.string());
         }
     }
+
+    // produce JSON for overridding system and user configuration;
+    // Options here need to match all options available for system and user settings
+    command_line_settings = json({}); // needs to have empty JSON at least in case of no command line options
+    for (const auto& key : int_keys_map) {
+        if (result.count(key.first)) {
+            command_line_settings[key.first] = result[key.first].as<int>();
```

Comment:
What happens here if the value can't be converted to a number? And what string values are valid for booleans?

Reply:
cxxopts takes care of that. If you use the wrong type of argument it should just exit